packages/completer/src/default/inlinehistoryprovider.ts

```diff
@@ -117,7 +122,7 @@ export class HistoryInlineCompletionProvider
         output: false,
         raw: true,
         hist_access_type: 'search',
-        pattern: linePrefix + '*',
+        pattern: linePrefix + '*' + suffix + '*',
```

Comment:
If there is not suffix it becomes `linePrefix + '**'. I am not sure if this will behave as expected, I would only add the second `*` if suffix is indeed there.